packages/block-library/src/post-template/index.php

```diff
@@ -78,24 +78,22 @@ function render_block_core_post_template( $attributes, $content, $block ) {
 	while ( $query->have_posts() ) {
 		$query->the_post();
 
-		// Get an instance of the current Post Template block.
-		$block_instance = $block->parsed_block;
-
 		// Set the block name to one that does not correspond to an existing registered block.
 		// This ensures that for the inner instances of the Post Template block, we do not render any block supports.
-		$block_instance['blockName'] = 'core/null';
-
+		$block->name = 'core/null';
+
+		$post_id              = get_the_ID();
+		$post_type            = get_post_type();
+		$filter_block_context = static function( $context ) use ( $post_id, $post_type ) {
+			$context['postType'] = $post_type;
+			$context['postId']   = $post_id;
+			return $context;
+		};
+		add_filter( 'render_block_context', $filter_block_context );
 		// Render the inner blocks of the Post Template block with `dynamic` set to `false` to prevent calling
 		// `render_callback` and ensure that no wrapper markup is included.
-		$block_content = (
-			new WP_Block(
-				$block_instance,
-				array(
-					'postType' => get_post_type(),
-					'postId'   => get_the_ID(),
-				)
-			)
-		)->render( array( 'dynamic' => false ) );
+		$block_content = $block->render( array( 'dynamic' => false ) );
```

Comment:
I left a similar comment under the PR that updates the Comment Template block. Have you considered passing the original context `$block->context` together with the locally modified context? I guess both approaches will work. The only question is why we landed on using the approach with a separate instance of `WP_Block` in this block. @ntsekouras, any insights we could miss?

Reply:
>The only question is why we landed on using the approach with a separate instance of WP_Block in this block. @ntsekouras, any insights we could miss?

I'm not really sure, but maybe even the `render_block_context` filter wasn't even in code yet. It seems both Query and this filter were introduced in the same WP version(5.5). There is a [discussion](https://github.com/WordPress/gutenberg/pull/22199/files#r423707686) about the context in the original PR, but nothing regarding the new instance.

Reply:
I'm still not sure why the issue with the layout classes is happening, but the culprit is definitely this change. Logging out `$block_content` shows the classes are already present in the nested block. This happens independently of whether the block in question supports layout or not: I can see the classes locally being added to Post Title.

Reply:
My current (and somewhat uninformed) hunch is that prior to this PR, we were [setting the block name to `core/null` for the copy of the block](https://github.com/WordPress/gutenberg/pull/50313/files#diff-4657bc4ddd88ab4cf8b48ad35b2fcf177edd40040c5976e2a3f9b89b52dd2bc7L86) that we created on the fly, whereas now, we're [setting it directly on the block itself](https://github.com/WordPress/gutenberg/pull/50313/files#diff-4657bc4ddd88ab4cf8b48ad35b2fcf177edd40040c5976e2a3f9b89b52dd2bc7R83). Maybe setting the block name back to its original value after [rendering](https://github.com/WordPress/gutenberg/pull/50313/files#diff-4657bc4ddd88ab4cf8b48ad35b2fcf177edd40040c5976e2a3f9b89b52dd2bc7R95) could fix this. I'll give that a try.

Reply:
Ah, I think I was slightly off. Judging by [this comment](https://github.com/WordPress/gutenberg/pull/50313/files#diff-4657bc4ddd88ab4cf8b48ad35b2fcf177edd40040c5976e2a3f9b89b52dd2bc7R81-R82)

```php
// Set the block name to one that does not correspond to an existing registered block.
// This ensures that for the inner instances of the Post Template block, we do not render any block supports.
```

...the whole point of setting the block name to `core/null` is to prevent the application of block supports -- which would lead to the duplication of class names that we're seeing here. So it's basically not working as it's supposed to.

My new hunch is that setting _only_ the `->name` isn't enough: Previously, we were setting the parsed block's `blockName` field, which is used by the `WP_Block` constructor to [set the `->name`](https://github.com/WordPress/wordpress-develop/blob/129843e1497cb21ea241eafd3df9b65c9394f66e/src/wp-includes/class-wp-block.php#L122) _but also_ [the `->block_type`](https://github.com/WordPress/wordpress-develop/blob/129843e1497cb21ea241eafd3df9b65c9394f66e/src/wp-includes/class-wp-block.php#L130).

So I think it's the latter that's missing here.